c++/src/kj/compat/tls.c++

```diff
@@ -80,6 +90,31 @@ void throwOpensslError() {
   kj::throwFatalException(getOpensslError());
 }
 
+#ifdef _WIN32
+// Adds windows CA store certificates to OpenSSL's ca store
+// Taken from https://stackoverflow.com/questions/9507184/can-openssl-on-windows-use-the-system-certificate-store
+void updateOpenSSLCAStoreWithWindowsCertificates(SSL_CTX* ctx) {
+  X509_STORE* store = SSL_CTX_get_cert_store(ctx);
+  if (store == nullptr) {
+    throwOpensslError();
+  }
+  HCERTSTORE hStore = CertOpenSystemStoreW(NULL, L"ROOT");
+  KJ_ASSERT(hStore != nullptr, "Failed to open system certificate store");
+  KJ_DEFER(CertCloseStore(hStore, 0));
```

Comment:
> Oh, also `CertOpenSystemStoreW` should probably check errors:
> 
> ```
>   HCERTSTORE hStore;
>   KJ_WIN32(hStore = CertOpenSystemStoreW(NULL, L"ROOT"));
> ```

I checked errors with `KJ_ASSERT(hStore != nullptr, "Failed to open system certificate store");` but the `KJ_WIN32` approach is indeed much nicer

Reply:
> `CertCloseStore` can fail, so we should wrap it in `KJ_WIN32`, which is the Windows equivalent of `KJ_SYSCALL`:
> 
> Meanwhile, according to docs, `CertFreeCertificateContext` always returns true... so I guess it doesn't really need it?
> 
> `CertEnumCertificatesInStore`, for its part, is a bit ambiguous... we expect it to return NULL at the end of the list, but it might also return NULL to indicate an error. Annoyingly, there are in fact two different error codes that might indicate end? Ugh I guess we can just skip checking the code and always assume null means end.

Yeah honestly I considered making this entire function just return early on failure without exceptions altogether just because if windows for some odd reason fails any of these calls we should probably just ignore and move on.
Eventually decided to do a half-assed attempt at error checking rather than none but I appreciate your comments trying to actually make this correct.